packages/cli/src/models/files/ZosNetworkFile.ts

```diff
@@ -0,0 +1,365 @@
+
+  get packageAddress(): string {
+    return this.package.address;
+  }
+
+  get provider(): any {
+    return this.data.provider || {};
+  }
+
+  get providerAddress(): string {
+    return this.provider.address;
+  }
+
+  get version(): string {
+    return this.data.version;
+  }
+
+  get frozen(): boolean {
+    return this.data.frozen;
+  }
+
+  get contracts(): any[] {
+    return this.data.contracts || {};
+  }
+
+  get contractAliases(): string[] {
+    return Object.keys(this.contracts);
+  }
+
+  get isLightweight(): boolean {
+    return this.packageFile.isLightweight;
+  }
+
+  get solidityLibs(): any[] {
+    return this.data.solidityLibs || {};
+  }
+
+  public addSolidityLib(libName: string, instance: any): void {
+    this.data.solidityLibs[libName] = {
+      address: instance.address,
+      constructorCode: constructorCode(instance),
+      bodyBytecodeHash: bytecodeDigest(bodyCode(instance)),
+      localBytecodeHash: bytecodeDigest(instance.constructor.bytecode),
+      deployedBytecodeHash: bytecodeDigest(instance.constructor.binary)
+    };
+  }
+
+  public unsetSolidityLib(libName: string): void {
+    delete this.data.solidityLibs[libName];
+  }
+
+  public setSolidityLib(alias: string, value: any): void {
+    if (!this.data.solidityLibs) this.data.solidityLibs = {};
+    this.data.solidityLibs[alias] = value;
+  }
+
+  public solidityLib(libName: string): any {
+    return this.data.solidityLibs[libName];
+  }
+
+  public getSolidityLibs(libs: any): any {
+    const { solidityLibs } = this.data;
+
+    return Object
+      .keys(solidityLibs)
+      .filter((libName) => libs.includes(libName))
+      .map((libName) => ({ libName, address: solidityLibs[libName].address }))
+      .reduce((someLibs, currentLib) => {
+        someLibs[currentLib.libName] = currentLib.address;
+        return someLibs;
+      }, {});
+  }
+
+  public hasSolidityLib(libName: string): boolean {
+    return !_.isEmpty(this.solidityLib(libName));
+  }
+
+  public solidityLibsMissing(libs: any): string[] {
+    return _.difference(Object.keys(this.solidityLibs), libs);
+  }
+
+  public getSolidityLibOrContract(aliasOrName: string): any {
+    return this.data.solidityLibs[aliasOrName] || this.data.contracts[aliasOrName];
+  }
+
+  public hasSolidityLibOrContract(aliasOrName: string): boolean {
+    return this.hasSolidityLib(aliasOrName) || this.hasContract(aliasOrName);
+  }
+
+  public updateImplementation(aliasOrName: string, fn: (...args: any[]) => any): void {
+    if (this.hasContract(aliasOrName)) this.data.contracts[aliasOrName] = fn(this.data.contracts[aliasOrName]);
+    else if (this.hasSolidityLib(aliasOrName)) this.data.solidityLibs[aliasOrName] = fn(this.data.solidityLibs[aliasOrName]);
+    else return;
+  }
+
+  get dependencies(): any[] {
+    return this.data.dependencies || {};
+  }
+
+  get dependenciesNames(): string[] {
+    return Object.keys(this.dependencies);
+  }
+
+  public getDependency(name: string): any | null {
```

Comment:
Another useful interfaces we can define here are `Dependencies` and `Dependency`.